dashboard/app/views/levels/levelGroup.js

```diff
@@ -20,86 +20,135 @@ function initLevelGroup(
       // show the Unsubmit button.
       $('.unsubmitButton').show();
     }
+
+    // Set the entire page background to be light grey.
+    $('.full_container').addClass("level_group_readonly");
   }
 
+
+  // Whenever an embedded level notifies us that the user has made a change,
+  // check for any changes in the response set, and if so, attempt to save
+  // these answers.  Saving is throttled to not occur more than once every 20
+  // seconds, and is done as soon as possible ("leading edge"), as well as at
+  // the end of a 20 second period if a change was made before then ("trailing
+  // edge").  Any pending throttled calls are cancelled when we go to a new page
+  // and save for that reason.
+
+  this.throttledSaveAnswers =
+    window.dashboard.utils.throttle(saveAnswers, 20 * 1000, {'leading': true, 'trailing': true});
+
+  var lastResponse = null;
+
+  window.answerChangedFn = function () {
+    var currentResponse = window.getResult().response;
+    if (lastResponse !== null && lastResponse !== currentResponse) {
+      this.throttledSaveAnswers();
+    }
+    lastResponse = currentResponse;
+  };
+
   /**
    * Construct an array of all the level results. When submitted it's something
    * like this:
    *
-   * [{"level_id":1977,"result":"0"},{"level_id":2007,"result":"3"},{"level_id":1939,"result":"2,1"}]
+   * [{"level_id":1977,"result":"0","valid":true},
+   *  {"level_id":2007,"result":"-1","valid":false},
+   *  {"level_id":1939,"result":"2,1","valid":true}]
```

Comment:
I think the updated structure is:
```json
{"1977":{"result":"0","valid":true},"2007":{"result":"-1","valid":false},"1939":{"result":"2,1","valid":true}}
```
(My fault for not updating in 0cd576d35ad1c2bff5900129241687a0102a0fea.)